src/Microsoft.Health.Fhir.Core/Configs/CoreFeatureConfiguration.cs

```diff
@@ -56,5 +56,10 @@ public class CoreFeatureConfiguration
         /// Maximum items allowed to be deleted when using Conditional Delete.
         /// </summary>
         public int ConditionalDeleteMaxItems { get; set; } = 100;
+
+        /// <summary>
+        /// Gets or sets a value whether capturing resource change data is enabled or not.
+        /// </summary>
+        public bool SupportsResourceChangeCapture { get; set; } = false;
```

Comment:
I actually think this should be tied to the DataStoreConfig since its is only planned as a SQL feature. Currently we use it straight from https://github.com/microsoft/healthcare-shared-components/blob/master/src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs

Reply:
From the SQL feature perspective, the SqlServerDataStoreConfiguration class might be a suitable place to have the property. But, this config is general and indicates whether an event feature is enabled in the Fhir Application. Also, it will be used to toggle the Cosmos DB's Change Feed feature. We might rename it to make it more generic. What you do think @ms-teli?

Reply:
@brendankowitz  There are some parallel discussion going on to leverage the event source pattern for more business requirements, like support change feed. The PM can give more info. Thus why we decide to make it more generic.

Reply:
I would move it to sql configuration as well.

If there is need to support this on all persistence levels we can always move it here, but that's in future.

Reply:
I think that the SupportsResourceChangeCapture config is an application-level configuration. If we move the config to the SqlServerDataStoreConfiguration class based on the SQL feature perspective like CDC, we need another config value that indicates whether an event feature is enabled in the FhirApplication in the appsetting.json. It leads to an issue managing two config values. For simplicity, we are using one application-level config which also toggles a SQL level feature.

Reply:
If it's a application level feature, why it works only for SQL and doesn't do anything for Cosmos?

If you move to CDC I expect complete rewrite of current functionality and drop all sql changes made in this PR.

If you want to have grace period, and switch back and forth, yes, it need second feature flag which would indicate implmentation, or you can always change type of this feauture from boolen to enum: `None,CDC,CustomTable`